samplesheets/views_api.py

```diff
@@ -137,6 +147,317 @@ def post(self, request, *args, **kwargs):
+
+    def post(self, request, *args, **kwargs):
+        """POST request for updating an iRODS data request"""
+        if not self.check_irods_permissions(request):
+            raise PermissionDenied('Insufficient permissions')
+
+        irods_request = IrodsDataRequest.objects.filter(
+            sodar_uuid=self.kwargs.get('irodsdatarequest')
+        ).first()
+        serializer = IrodsRequestSerializer(
+            irods_request, data=request.data, partial=True
+        )
+        if serializer.is_valid():
+            serializer.save()
+        else:
+            raise ValidationError('Invalid data: {}'.format(serializer.errors))
+
+        irods_request = IrodsDataRequest.objects.filter(
+            sodar_uuid=serializer.data.get('sodar_uuid')
+        ).first()
+        # Create timeline event
+        self.add_tl_update(irods_request)
+
+        return Response(
+            {
+                'detail': 'iRODS data request updated',
+                'request': irods_request.sodar_uuid,
+            },
+            status=status.HTTP_200_OK,
+        )
+
+
+class IrodsRequestDeleteAPIView(
+    IrodsRequestModifyMixin, SODARAPIBaseProjectMixin, APIView
+):
+    """
+    Delete an iRODS data request for a project.
+
+    **URL:** ``/samplesheets/api/irods/request/delete/{IrodsDataRequest.sodar_uuid}``
+
+    **Methods:** ``DELETE``
+    """
+
+    http_method_names = ['delete']
+    # permission_required = 'samplesheets.edit_sheet'
+    permission_classes = (IsAuthenticated,)
+
+    def get_object(self):
+        """Override to allow POST requests"""
+        return IrodsDataRequest.objects.get(
+            sodar_uuid=self.kwargs.get('irodsdatarequest')
+        )
+
+    def check_irods_permissions(self, request):
+        """Override to allow POST requests without CSRF token"""
+        if (
+            request.user.is_superuser
+            or self.get_project().is_owner(request.user)
+            or request.user.has_perm(
+                'samplesheets.manage_sheet', self.get_object().project
+            )
+            or request.user == self.get_object().user
+        ):
+            return True
+        return False
+
+    def delete(self, request, *args, **kwargs):
+        """DELETE request for deleting an iRODS data request"""
+        if not self.check_irods_permissions(request):
+            raise PermissionDenied('Insufficient permissions')
+
+        ex_msg = 'Deleting' + IRODS_EX_MSG
+        irods_request = IrodsDataRequest.objects.filter(
+            sodar_uuid=self.kwargs.get('irodsdatarequest')
+        ).first()
+        try:
+            irods_request.delete()
+        except Exception as ex:
+            raise APIException('{} {}'.format(ex_msg, ex))
+
+        # Add timeline event
+        self.add_tl_delete(irods_request)
+        # Handle project alerts
+        self.handle_alerts_deactivate(irods_request)
+        return Response(
+            {
+                'detail': 'iRODS data request deleted',
```

Comment:
I think here we can again omit some unneeded line breaks by removing the comma and placing the dict on one line.